File: 3267-find-longest-special-substring-that-occurs-thrice-i/3267-find-longest-special-substring-that-occurs-thrice-i.py
class Solution(object):
    def maximumLength(self, s):
        n, mp = len(s), {}

        for i in range(n):
            curr = s[i]
            if curr in mp:
                mp[curr] += 1
            else:
                mp[curr] = 1

            for j in range(i, n - 1):
                if s[j] == s[j+1]:
                    curr += s[j]
                    if curr in mp:
                        mp[curr] += 1
                    else:
                        mp[curr] = 1
                else:
                    break
        
        ans = 0
        for key, value in mp.items():
            if value >= 3:
                ans = max(ans, len(key))
        
        return ans if ans != 0 else -1
        

File: 3267-find-longest-special-substring-that-occurs-thrice-i/3267-find-longest-special-substring-that-occurs-thrice-i.cpp
class Solution {
public:
    int maximumLength(string s) {
        unordered_map<string ,int> mp;
        int n = s.size();

        for(int i = 0; i<n; i++) {
            string str = "";
            str += s[i];
            mp[str]++;

            for(int j = i; j<n-1; j++) {
                if(s[j] == s[j+1]) {
                    str += s[j];
                    mp[str]++;
                } else break;
            }
        }

        int ans = 0;
        for(auto it : mp) {
            if(it.second >= 3) {
                ans = max(ans, (int)it.first.size());
            }
        }

        return ans == 0 ? -1 : ans;
    }
};